src/AnonymizationStep/AnonymizedResultsTable.tsx

```diff
@@ -21,54 +23,97 @@ type TableRowData = RowData & {
 
 // Columns
 
+function cellDefaultTooltip(displayValue: string) {
+  return <Tooltip title={displayValue}>{displayValue}</Tooltip>;
+}
+
+function cellNullTooltip() {
+  return (
+    <Tooltip title="NULL">
+      <i>NULL</i>
+    </Tooltip>
+  );
+}
+
+function cellNumericGeneralizedTooltip(binSize: number, v: number) {
+  const displayValue = v.toString();
+  const tooltip = `[${v}; ${v + binSize})`;
+  return <Tooltip title={tooltip}>{displayValue}</Tooltip>;
+}
+
+function cellTooltip(bucketColumn: BucketColumn | undefined, v: number | boolean | string) {
+  if (
+    bucketColumn &&
+    (bucketColumn.type === 'integer' || bucketColumn.type === 'real') &&
+    bucketColumn.generalization &&
+    bucketColumn.generalization.binSize
+  ) {
+    return cellNumericGeneralizedTooltip(bucketColumn.generalization.binSize, v as number);
+  } else {
+    return cellDefaultTooltip(v.toString());
+  }
+}
+
 function renderValue(v: Value) {
+  // with `ellipsis: { showTitle: false }` set for the column, we're disabling the default tooltip
+  // so that the styling of tooltips is more consistent. This forces us to provide Tooltip in all cases
+  // even if the tooltip shown is same as the cell value.
   if (v === null) {
-    return <i>NULL</i>;
+    return cellNullTooltip();
   } else {
-    return v.toString();
+    return cellDefaultTooltip(v.toString());
   }
 }
 
+function buildRenderTooltipValue(column: AnonymizedResultColumn, bucketColumns: BucketColumn[]) {
+  const bucketColumn = bucketColumns.find((c) => c.name === column.name);
+  return (v: Value) => {
+    // see note on `ellipsis` above
+    if (v === null) {
+      return cellNullTooltip();
+    } else {
+      return cellTooltip(bucketColumn, v);
+    }
+  };
+}
+
 function renderLowCountValue(v: Value) {
-  return v === null ? '-' : v.toString();
+  // see note on `ellipsis` above
+  return cellDefaultTooltip(v === null ? '-' : v.toString());
 }
 
 function renderRelativeNoiseValue(v: Value) {
-  return v === null ? '-' : formatPercentage(v as number);
+  // see note on `ellipsis` above
+  return cellDefaultTooltip(v === null ? '-' : formatPercentage(v as number));
 }
 
-function makeColumnData(
-  title: string,
-  dataIndex: string,
-  type: ColumnType,
-  render: (v: Value) => React.ReactNode = renderValue,
-) {
+function makeColumnData(title: string, dataIndex: string, type: ColumnType, render: (v: Value) => React.ReactNode) {
   return {
     title,
     dataIndex,
     render,
     sorter: columnSorter(type, dataIndex),
-    ellipsis: true,
+    ellipsis: { showTitle: false },
```

Comment:
Do we lose tooltip in column titles now?

Reply:
nope. If this was left at `true` we'd have 2 tooltips for the columns that have the generalization (and with different style). 

At current state, we have always a single tooltip, with style consistent per table. See comment in the [original PR](https://github.com/diffix/desktop/pull/253#issue-780588667)

Reply:
Oh, you've edited your comment. So I think this `showTitle` setting has a confusing name. It's about tooltips in the `ellipsis`ed cells